libraries/Wire/src/utility/twi.c

```diff
@@ -846,9 +846,10 @@ i2c_status_e i2c_slave_write_IT(i2c_t *obj, uint8_t *data, uint16_t size)
   } else {
     // Check the communication status
     for (i = 0; i < size; i++) {
-      obj->i2cTxRxBuffer[i] = *(data + i);
+      obj->i2cTxRxBuffer[obj->i2cTxRxBufferSize + i] = *(data + i);
     }
 
+    size += obj->i2cTxRxBufferSize;
     obj->i2cTxRxBufferSize = size;
```

Comment:
What I meant was to merge both lines, as there is no need to update `size`:
```
obj->i2cTxRxBufferSize += size;
```

Reply:
Will be ready for merge when fixed in the first commit.

Reply:
now I understood what you meant!